Src/Particle/AMReX_Particle.H

```diff
@@ -55,7 +55,7 @@ struct Particle
     */
     union im_t
     {
-      struct {int id; int cpu;};
+      int ids[2];
```

Comment:
Maybe we should remove the union entirely.  Technically it's UB, if one tries to access both `ids` and `arr`.

Reply:
I agree and we should do the same for the real data, but I'd like that to be a follow-up PR since I'll also have to deal with codes using `m_rdata.pos` directly.